addons/nuxeo-platform-3d/nuxeo-platform-3d-core/src/main/java/org/nuxeo/ecm/platform/threed/service/ThreeDBatchUpdateWork.java

```diff
@@ -92,6 +91,29 @@ public boolean isIdempotent() {
         return false;
     }
 
+    @Override
+    public boolean isGroupJoin() {
+        // This is a GroupJoin work with a trigger that can be used on the last work execution
+        return true;
+    }
+
+    @Override
+    public String getPartitionKey() {
+        return computeIdPrefix(repositoryName, docId);
+    }
+
+    @Override
+    public void onGroupJoinCompletion() {
+        if ("DONE".equals(getStatus())) {
+            try {
+                openSystemSession();
+                fireThreeDConversionsDoneEvent(session.getDocument(new IdRef(docId)));
+            } finally {
+                cleanUp(true, null);
+            }
+        }
+    }
```

Comment:
i am wondering as you call the `cleanUp` with a ` true` as a first pram (means just closing the session [here](https://github.com/nuxeo/nuxeo/blob/f0931542b914da30307a3be4225a4528dd905486/nuxeo-core/nuxeo-core-event/src/main/java/org/nuxeo/ecm/core/work/AbstractWork.java#L542)) what about using the try final ressource syntax :

```
try  (CloseableCoreSession s = CoreInstance.openCoreSessionSystem(repositoryName, originatingUsername)){
                fireThreeDConversionsDoneEvent(s.getDocument(new IdRef(docId)));
 }
```

Reply:
Yes you might be right but I prefer to follow the convention used in this class which is openSystemSession/cleanUp.

Reply:
yep i see :),